### PR TITLE
Clang compiler

### PR DESCRIPTION
**Pull request by stigrj**
Need to fix issues with Clang compiler in order to build functionally on Ubuntu-18.10:
- [x] Update autocmake
- [x] Add Clang compiler options
- [x] Add Clang build on Travis
- [x] Ready to go

### CODE REVIEW

File: tests/scalingfactor/integral.cpp [hunk 1]
```diff
@@ -23,6 +23,7 @@ constexpr auto alpha_gen(const double &sigma) {
     return std::pow(1.0/std::sqrt(2.0*pi*std::pow(sigma, 2.0)), D);
 }
 
+template<int D>
```
[the remark this answers is not in the document]
Comment (stigrj) [reply tests/scalingfactor/integral.cpp]: Just checking if you pay attention :wink: Clang complained it wasn't a `constexpr`. Adding a template silenced the compiler and made the three helper functions consistent and nice :smile_cat:

[the remark this answers is not in the document]
Comment (robertodr) [reply tests/scalingfactor/integral.cpp]: Maybe it's a sign it shouldn't be `constexpr`?

**Comment by stigrj** [comment]
The Clang build on Travis segfaults in `release` mode for some reason (version 3.8.0 it seems, see below).

Also, with
```
packages:
  - clang
```
we get
```
$ ${CXX_COMPILER} --version
clang version 7.0.0 (tags/RELEASE_700/final)
```
but CMake reports
```
-- The CXX compiler identification is Clang 3.8.0
```
However, with
```
packages:
  - clang-5.0
```
we still get
```
$ ${CXX_COMPILER} --version
clang version 7.0.0 (tags/RELEASE_700/final)
```
but this time CMake also reports v7.0.0:
```
-- The CXX compiler identification is Clang 7.0.0
```
This version compiles in `release` mode but the `-omp` tests fail due to missing `libomp`. My head hurts.

**Comment by robertodr** [comment]
I never quite understood Clang versioning, to be honest. And this is even without taking the Mac version into consideration. It might be that CMake is reporting the version of LLVM on which the front-end is based. Anyway, on the topic of OpenMP: until very recently it was experimental in LLVM and most likely the package on Ubuntu hasn't caught up yet.

**Comment by stigrj** [comment]
Things worked a little better in MRChem so I have tried to copy the Travis setup. The OpenMP is still failing, though, even if it works on my laptop. For the time being I'm fine with only serial Clang build on Travis.

**Comment by stigrj** [comment]
I think one is as ready as it gets

**Pull request merged**